app.rb

```diff
@@ -0,0 +1,100 @@
+require_relative 'book'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'person'
+require_relative 'rental'
+
+def option1
+  puts '** Books **'
+  ObjectSpace.each_object(Book) do |obj|
+    puts "Title: #{obj.title} Author: #{obj.author}"
+  end
+end
+
+def option2
+  puts '** Teachers **'
+  ObjectSpace.each_object(Teacher) do |obj|
+    puts "Id: #{obj.id} Name: #{obj.name} Age: #{obj.age} Specialization: #{obj.specialization}"
+  end
+  puts '** Students **'
+  ObjectSpace.each_object(Student) do |obj|
+    puts "Id: #{obj.id} Name: #{obj.name} Age: #{obj.age}"
+  end
+end
+
+def opcion3
+  puts 'Do you want to create a student (1) or a teacher (2) [Input the number]:'
+  target = gets.chomp
+  option_student if target == '1'
+  option_teacher if target == '2'
+end
+
+def option4
+  puts 'Title:'
+  title = gets.chomp
+  puts 'Author:'
+  author = gets.chomp
+
+  Book.new(title, author)
+  puts 'Book created successfully'
+end
+
+def option5
+  book_array = ObjectSpace.each_object(Book).to_a
+  person_array = ObjectSpace.each_object(Person).to_a
+  if book_array.empty? || person_array.empty?
+    puts 'Book or Person list are empty...'
+  else
+    puts 'Select a Book Number from the following list'
+    book_array.each_with_index { |book, index| puts "#{index + 1}. #{book.title} - #{book.author}" }
+    puts 'Book Number:'
+    book_number = gets.chomp.to_i
+    puts 'Select a Person Number from the following list (not id)'
+    person_array.each_with_index do |person, index|
+      puts "#{index + 1}. #{person.class.name} #{person.name} - id: #{person.id}"
+    end
+    puts 'Person Number:'
+    person_number = gets.chomp.to_i
+    puts 'Date (dd-mm-yy):'
+    date = gets.chomp
+    Rental.new(date, person_array[person_number - 1], book_array[book_number - 1])
+    puts 'Rental created successfully'
+  end
+end
```

Comment:
- [ ] Please don't forget that when creating a rental, the information returned to the user needs to look similar to what was done in the Example video so as to fulfill the project requirements and also so the `ID` of a person is easily accessible as I mentioned before :wink: 
- [ ] After this, please ensure that users are able to `List all rentals for a given person's id` :+1: 

| Your UI | The Example Video |
|--------|--------|
| ![image](https://github.com/haftamudesta/school-library/assets/102699958/cde2307f-cab6-447b-a4f2-30eb69e6907c) | ![image](https://github.com/haftamudesta/school-library/assets/102699958/dcb7e140-a1ae-4085-a8bc-50e974e2ef83)|